### DISCUSSION TITLE
Using it on website

### DISCUSSION THREAD

**Discussion by kaushiksampath77** [PromtEngineer/localGPT#19] [Q&A] [answered]
How can I used LocalGPT on a website to provide customer support chat system using private PDFs or DOC files?

**Comment by jnfarooq** [comment, marked as answer]
In theory, you could. But it has to be through an API. You will also want to potentially look at changing the LLM being used. Here its using Vicuna-7B which is based on Llama.

**Comment by kinthaiofficial** [comment]
Running localGPT on a website (web-accessible) rather than purely local adds some interesting constraints:

**Option 1: FastAPI wrapper** (simplest):
```python
from fastapi import FastAPI
from fastapi.responses import StreamingResponse
from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler

app = FastAPI()

@app.post("/chat")
async def chat(request: ChatRequest):
    async def generate():
        for token in qa_chain.stream(request.message):
            yield f"data: {token}\n\n"
    return StreamingResponse(generate(), media_type="text/event-stream")
```

**Option 2: Gradio** (fast to build, less control):
```python
import gradio as gr

def chat(message, history):
    return qa_chain(message)["result"]

demo = gr.ChatInterface(chat)
demo.launch(server_name="0.0.0.0", server_port=7860)
```

**Security considerations for web-exposed deployment**:
- Rate limiting (each query can be expensive)
- Authentication (don't want anyone hitting your server)
- Context isolation between users (don't leak one user's conversation to another)
- Consider ngrok or Cloudflare Tunnel rather than exposing directly

**Multi-user consideration**: localGPT uses a single model instance. Concurrent requests will queue. For a shared web deployment, you'd want to look at a proper serving layer like llama.cpp server (supports concurrent requests up to its configured limit).

What's your deployment target — personal use accessible from multiple devices, or shared with other users?
